### PR TITLE
Rename sections in Expectations Details view

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

Currently, the structure of the view mirrors the structure of the check itself, using sections named "Expectations," "Values," and "Facts" for expect type expectations, and "Expectations" and "Facts" for expect_same type expectations.

These section names might not be immediately clear to users who are not familiar with the checks DSL.

## For expect type expectations:

- "Expectations" section will be renamed to "Evaluation Results"
- "Values" section will be renamed to "Expected Values"
- "Facts" section will be renamed to "Gathered Facts"
## For expect_same type expectations:

- "Expectations" section will be renamed to "Evaluation Results"
- "Facts" section will remain unchanged as "Gathered Facts"

### ExpectationsResults

[ExpectationsResults.webm](https://github.com/trento-project/web/assets/54111255/526ee93f-f50e-43b4-95ec-56bd0995b926)

### ExpectedValues

[ExpectedValues.webm](https://github.com/trento-project/web/assets/54111255/30ac0b17-12d1-4698-9689-d4f3232002c4)

### Gathered Facts

[GatheredFacts.webm](https://github.com/trento-project/web/assets/54111255/073f522c-8b60-4879-a7e4-fdf6eb457da4)

### CODE REVIEW

File: assets/js/components/ExecutionResults/CheckResultDetail/ExpectationsResults.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    results: [
+      { name: faker.animal.bear(), return_value: true },
+      { name: faker.animal.bear(), return_value: false },
+    ],
+    isError: false,
+    errorMessage: 'An error occurred',
+  },
+};
+
+export function Default(args) {
```
Comment (nelsonkopliku) [inline assets/js/components/ExecutionResults/CheckResultDetail/ExpectationsResults.stories.jsx:19]: Fow new stories we're aiming to follow the way of writing them that has been introduced in the latest versions of storybook https://storybook.js.org/docs/react/writing-stories/introduction

See [Accordion](https://github.com/trento-project/web/blob/main/assets/js/components/Accordion/Accordion.stories.jsx) and [CheckResultDetail](https://github.com/trento-project/web/blob/main/assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx) and https://github.com/trento-project/web/pull/1428#discussion_r1200129544

**Review by nelsonkopliku — CHANGES_REQUESTED** [review]
Super! Just a tiny change in the way we write stories.

File: assets/js/components/ExecutionResults/CheckResultDetail/ExpectationsResults.stories.jsx (hunk 1 above)
Comment (EMaksy) [reply assets/js/components/ExecutionResults/CheckResultDetail/ExpectationsResults.stories.jsx:19]: Oh thanks for reminding me!  :+1:

**Review by nelsonkopliku — APPROVED** [review]
aw yeah :shipit:

**Review by dottorblaster — APPROVED** [review]
:shipit:

**Pull request merged**